### PR TITLE
allowlist: Add partners operator CPU usage metric

### PR DESCRIPTION
**Pull request by machacekondra**
This PR add new partners:restricted_use_cores:sum metric.

This metric is used by our partners to report the CPU usage of their
operators.

* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: Documentation/data-collection.md [hunk 1]
```diff
@@ -332,6 +332,9 @@ data:
     # (@openshift/team-build-api, OpenShift Build API Team) openshift:build_by_strategy:sum measures total number of builds on a cluster, aggregated by build strategy.
     - '{__name__="openshift:build_by_strategy:sum"}'
     #
+    # (partners metrics) The number of the cores used by the partner operator.
```
Comment (simonpasquier) [inline Documentation/data-collection.md:335]: we ask for a GitHub team handle + name here (see examples above).

Comment (machacekondra) [reply Documentation/data-collection.md:335]: This is special case, where the metric is emitted by ISV partner operator, so there is no team responsible for the metric. What do you suggest to put here?

Comment (bwplotka) [reply Documentation/data-collection.md:335]: Let's talk about consequences. If this metric will start exposing millions series suddenly, who we should talk to fix this or to notify that we are removing this metric due to abuse?

**Pull request closed without merging**